benchopt/base.py

```diff
@@ -443,6 +443,37 @@ def save_final_results(self, **solver_result):
         """
         pass
 
+    def format_objective_dict(self, objective_dict):
+        """Format the output of the objective.
+
+        This will prefix all keys in the dictionary with `objective_`
+        to make the objective part of the results clear.
+
+        Parameters
+        ----------
+        objective_dict: dict
+            The output of the objective function, which should be a dictionary
+            not containing the key 'name'.
+
+        Returns
+        -------
+        objective_dict : dict
+            The formatted objective to include in the DataFrame.
+        """
+
+        if not isinstance(objective_dict, dict):
+            raise ValueError(
+                "The output of the objective should be a list of dictionary "
```

Comment:
This is a bit misleading as we allow a single float or lists of dicts ; also I'd remove the 'name' key part of the msg as is not related to checking if it's a dict or not

Reply:
Anyways, because of the not isinstance(list, dict) test performed in call and what we do here https://github.com/benchopt/benchopt/pull/778/files#diff-16b4460abb25e65fd02f1ded976a13c1ab21cbf0b69e1eaa586ca38c7dee12daR493  I'm not sure there are many cases we end up with a non dict over there 

The only case I see would be when objective_output is a list of whatever.

Reply:
Yes, we only end up here if we have an element of the list that is not a dict so we should point that we should either have a dict or a list of dict.